In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [5]:
players <- read_csv("players.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [6]:
sessions <- read_csv("sessions.csv")

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

(1) Data Description: The players contains 196 observations and 7 variables, including exprience,hashedEmail,name,gender,played_hours,age and subscribe. The key reposonse variable is subscribe, it is the logical variable indicating whether the player subscribed to a game-related newsletter. Predictor variables include numerical features like age and played_hours. The dataset may contain missing or inconsistent entries, and the subscription rate may be unbalanced. The sessions contains 1535 observations and 5 variables: hashedEmail,start_time,end_time,original_start_time,original_end_time. It records session level activity, which can be used to calculate total playtime, session frequency and active hours. The two dataset both have hashedEmail, so they can be linked by using this variable. 

(2)Questions: Broad Question: what player characteristics and behaviors are most predictable of subscribing to a game-related newsletter? 

Specific question: Can player characteristics like age, experience and total played hours predict whether a player subscribes to the newsletter? 

The analysis will use the players.csv dataset and may be combined with the aggregated session information from the sessions.csv file. The goal is to determine which features are closely related to the likelihood of subscription.

In [7]:
library(tidyverse)
players_var <-players|>
              select(experience, gender, Age, played_hours, subscribe)
players_var

experience gender            Age played_hours subscribe
1   Pro        Male               9  30.3          TRUE    
2   Veteran    Male              17   3.8          TRUE    
3   Veteran    Male              17   0.0         FALSE    
4   Amateur    Female            21   0.7          TRUE    
5   Regular    Male              21   0.1          TRUE    
6   Amateur    Female            17   0.0          TRUE    
7   Regular    Female            19   0.0          TRUE    
8   Amateur    Male              21   0.0         FALSE    
9   Amateur    Male              47   0.1          TRUE    
10  Veteran    Female            22   0.0          TRUE    
11  Veteran    Female            23   1.6          TRUE    
12  Pro        Male              17   0.0          TRUE    
13  Beginner   Female            25   1.5          TRUE    
14  Amateur    Male              22   0.2          TRUE    
15  Veteran    Non-binary        17   0.0          TRUE    
16  Amateur    Male              22   0.0         FALSE    
17  Beginner   Female            17   0.0          TRUE    
18  Amateur    Female            17  48.4          TRUE    
19  Amateur    Male              17   0.5          TRUE    
20  Regular    Male              19   0.6          TRUE    
21  Regular    Male              58   0.3          TRUE    
22  Amateur    Female            17   0.1         FALSE    
23  Beginner   Male              17   1.0          TRUE    
24  Beginner   Male              17   0.0          TRUE    
25  Amateur    Female            21   0.7          TRUE    
26  Regular    Female            28   0.6          TRUE    
27  Regular    Male              17   0.0          TRUE    
28  Veteran    Male              23   0.0         FALSE    
29  Amateur    Male              23   1.8          TRUE    
30  Veteran    Male              18   0.1         FALSE    
⋮   ⋮          ⋮                 ⋮   ⋮            ⋮        
167 Amateur    Male              26   1.0         FALSE    
168 Beginner   Female            17   0.3         FALSE    
169 Veteran    Male              17   0.1          TRUE    
170 Veteran    Male              17   0.1         FALSE    
171 Pro        Male              21   0.4         FALSE    
172 Beginner   Male              32   1.8         FALSE    
173 Veteran    Agender           20   0.0          TRUE    
174 Regular    Male              50   0.0         FALSE    
175 Veteran    Male              17   0.0          TRUE    
176 Amateur    Male              17   0.3          TRUE    
177 Pro        Non-binary        17   0.0         FALSE    
178 Veteran    Non-binary        21   2.7         FALSE    
179 Amateur    Two-Spirited      17   0.4          TRUE    
180 Veteran    Non-binary        17   0.1          TRUE    
181 Amateur    Male              24   2.1         FALSE    
182 Amateur    Female            22   0.8          TRUE    
183 Pro        Male              17   0.2         FALSE    
184 Amateur    Male              22  32.0          TRUE    
185 Pro        Male              17   1.7         FALSE    
186 Regular    Male              18   0.1         FALSE    
187 Veteran    Female            44   0.1          TRUE    
188 Amateur    Male              17   0.0          TRUE    
189 Beginner   Female            17   0.0          TRUE    
190 Amateur    Male              17   0.0         FALSE    
191 Amateur    Male              20   0.0          TRUE    
192 Amateur    Female            17   0.0          TRUE    
193 Veteran    Male              22   0.3         FALSE    
194 Amateur    Prefer not to say 57   0.0         FALSE    
195 Amateur    Male              17   2.3         FALSE    
196 Pro        Other             NA   0.2          TRUE

In [8]:
library(tidyverse)
players_clean <- players_var |>
                summarise(mean_age = mean(Age,na.rm = TRUE),
                         mean_hours = mean(played_hours,na.rm = TRUE),
                         subscribe_rate = mean(subscribe, na.rm = TRUE))
players_clean

mean_age mean_hours subscribe_rate
1 21.13918 5.845918   0.7346939

In [ ]:
library(ggplot2)
players_age <- ggplot(players_var, aes(x = Age))+
                geom_histogram(binwidth = 5, color = "black")+
                labs(title = "Played Hours by Subscription Status",
                    x = "Age",
                    y = "Count")
players_age

Warning message:
“Removed 2 rows containing non-finite outside the scale range (`stat_bin()`).”
